apps/src/music/views/Generate.tsx

```diff
@@ -0,0 +1,195 @@
+      length: ['20', '30', '40'],
+    },
+  },
+  sounds: {
+    template: 'Please make a song with {sounds} sounds in order.',
+    options: {
+      sounds: ['2', '3', '4', '5'],
+    },
+  },
+  layers: {
+    template:
+      'Please make a song that layers {type1} and {type2} together. Around 10 measures is good.',
+    options: {
+      type1: ['bass', 'leads', 'beats', 'vocals'],
+      type2: ['bass', 'leads', 'beats', 'vocals'],
+    },
+  },
+};
+
+interface GenerateProps {}
+
+const Generate: React.FunctionComponent<GenerateProps> = () => {
+  const dispatch = useAppDispatch();
+
+  const packId = useAppSelector(state => state.music.packId);
+  const aiGenerateState = useAppSelector(state => state.music.aiGenerateState);
+
+  const library = MusicLibrary.getInstance();
+
+  const adlibOption = appConfig.getValue('ai-generate-adlib') as string;
+  const [adlibText, setAdlibText] = useState<string | undefined>(undefined);
+
+  const [text, setText] = useState(
+    'Please generate a fun song.  Between 18-20 measures is enough duration.  Use layering of sounds to make it exciting.'
+  );
+
+  const useText = adlibOption ? adlibText : text;
+
+  const sounds = library
+    ?.getFolderForFolderId(packId || 'indie')
+    ?.sounds?.map(sound => {
+      if (sound.type !== 'preview') {
+        return `${sound.src} (${sound.length} measures)`;
+      }
+    })
+    .filter(sound => sound !== undefined)
+    .join('", "');
+
+  const contextGenerateMusicPsuedocodeFromDescription = `Your job will be to generate psuedocode for a system that plays a song.  You'll be given a description of what to play, and then you should output code that generates the song to be played.  The psuedocode looks something like this:
+
+when_run
+  play "hiphop/drum_beat_808"
+  play "electro/drum_beat_hyper"
+  play_together
+    play "hiphop/drum_beat_808"
+    play "electro/drum_beat_hyper"
+  repeat 3
+    play "hiphop/drum_beat_808"
+    play "electro/drum_beat_hyper"
+
+Indenting is important.  In this example, when the code is run, it plays "hiphop/drum_beat_808" and then "electro/drum_beat_hyper".  Then it plays "electro_beat_808" and "electro/drum_beat_hyper" at the same time.  Then it plays the same thing three times: "hiphop/drum_beat_808" followed by "electro/drum_beat_hyper".
+
+Don't include any comments in the generated psuedocode.
+
+Note that each sound is actually 2 measures long.
+
+The valid sounds to use are: "${sounds}".  (The length of each sound is in parentheses.)  You can use any of these sounds in your psuedocode.  Each sound name gets the "${packId}/" prefix, so for example, "indie/drum_beat_808".
+`;
+
+  const generateSong = useCallback(() => {
+    console.log('starting ask');
+    dispatch(setAiGenerateState('generating'));
+    askAi(
+      `Here is the context:
+  ${contextGenerateMusicPsuedocodeFromDescription}
+
+  And here is the request:
+  ${useText}`
+    ).then(result => {
+      console.log(result[1].chatMessageText);
```

Comment:
Might be good to have some basic error handling here for the time being (mostly for dev visibility - `console.error` is fine) - we should check 1) if `result.length` is > 1 and 2) if `result[0]` or `result[1]` has a status other than `AiInteractionStatus.OK`

Tangentially, it would be nice for the API to just return the AI response or an error code rather than the list of messages, which is a very AI Chat-specific use case. But since we don't plan to call the AI in production, probably not necessary to refactor that right now.

Reply:
Added.